### PR TITLE
OZ-715: Introducing test profiles + renamed "goTo" to "navigateTo" routines.

### PR DESCRIPTION
**Pull request by kdaud**
Ticket: https://mekomsolutions.atlassian.net/browse/OZ-715

This PR adds ability to select preferred test profile (at runtime) that only runs a subset of the E2E test suite. Also renamed `goToDrugOrderForm() → navigateToDrugOrderForm()` and `goToLabOrderForm() → navigateToLabOrderForm()`

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      test_pro:
+        description: 'Running Ozone on Pro?'
+        required: true
+        default: 'true'
+        type: choice
+        options:
+          - true
+          - false
+      test_profile:
+        description: 'Choose test profile'
```
Comment (VaishSiddharth) [inline .github/workflows/e2e.yml:28]: Should we simplify this to `Choose Distro`
cc: @mks-d

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        uses: actions/cache@v4
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Install Playwright browsers
+        run: npx playwright install chromium
```
Comment (VaishSiddharth) [inline .github/workflows/e2e.yml:62]: should we add `--with-deps` flag?

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        run: |
+          npm run $TEST_PROFILE
+
+      - name: Upload report
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: playwright-report
+          path: playwright-report/
+          retention-days: 30
```
Comment (VaishSiddharth) [inline .github/workflows/e2e.yml:84]: A side note do we really need to retain these reports for 30 days? Why not reduce this to a week?
cc: @enyachoke

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (kdaud) [reply .github/workflows/e2e.yml:28]:
```code      
options:
        - ozone-pro
        - ozone-foss
        - openmrs-distro-his
  ```
@VaishSiddharth those are not actually distributions but rather profiles that run a subset of E2E test suite. Refer to https://github.com/ozone-his/ozone-e2e/pull/112/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R16-R20

**Comment by VaishSiddharth** [comment]
I have left some comments apart from that the PR looks good!

Also I was looking into the e2e tests functions and I could see that there are a lot of `delay` statements are present.
I think a better option will be to use [Playwright Auto-waiting](https://playwright.dev/docs/actionability) functionality. 
If we continue using `delay` statements I worry that the following will happen:
-  We will end up increasing the delay time with every PR to accommodate all scenarios where the UI load speeds are very slow.
-  These delays could reach to even minutes and hence our e2e tests eventually be slower on faster machines as well.

Having said that I suspect that we might not be able to remove all the `delay` statements but lets try to minimise them as much as we can.

This might require some refactoring so we might want to create a new ticket for this.

Please share your thoughts on this if any.
cc: @kdaud @mks-d @corneliouzbett @mks-d

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (kdaud) [reply .github/workflows/e2e.yml:62]: I prefer not to add the suggested flag to download only the necessary browser binaries

**Comment by kdaud** [comment]
> Also I was looking into the e2e tests functions and I could see that there are a lot of delay statements present.
I think a better approach will be to use [Playwright Auto-waiting](https://playwright.dev/docs/actionability) functionality.

We use Playwright's auto-waiting functionality in our tests, but there are cases where delays are added to ensure certain actions occur. For example, a 5-second delay is added after placing an order in O3 to allow synchronization with the lab and billing systems.

> but lets try to minimise them as much as we can.

I agree.

**Review by VaishSiddharth — APPROVED** [review]

**Pull request merged**